### DISCUSSION TITLE
Annotate on client + train on server

### DISCUSSION THREAD

**Discussion by andreped** [Project-MONAI/MONAILabel#1197] [General]
I have now returned back to MONAI Label for further testing, and we wanted to test it on a new application. Our goal is to be able to annotate locally through 3D Slicer, but train on a local server we control. The idea is basically to reduce the compute load on the local annotation machine by training on a server instead, which have access to a much larger GPU. We are annotating CT data.

Is that possible? If yes, are there any documentations on how to do that?

**Comment by tangy5** [comment]
Thanks for the question.
If I understand correctly here, you would like to use MONAI Label+3D Slicer for inference/annotate, but won't use MONAI Label for training (or iteratively train). 

If so, I suppose it's simple.

We are making some E2E tutorials/Doc for MONAI Label+3D Slicer use cases. It's pending to merge.
You can find these:
https://github.com/tangy5/tutorials/tree/monailabel_tutorial_series/monailabel#monai-label-end-to-end-tutorial-series

Especially this tutorial for Radiology app and use python script for config models/network. Inference pipeline till Section 2. And you can ignore section 3-4 for training and active learning.
https://github.com/tangy5/tutorials/blob/monailabel_tutorial_series/monailabel/monailabel_HelloWorld_radiology_3dslicer.ipynb

And MONAI Label is also supporting bundles, if you prefer to use `bundle` for loading models.
https://github.com/tangy5/tutorials/blob/monailabel_tutorial_series/monailabel/monailabel_monaibundle_3dslicer_multiorgan_seg.ipynb


You could only do inference step and ignore training, then after getting all annotated labels, you could train your data in many ways with local server. 

Thanks.

**Reply by andreped** [reply]
Thanks for the fast reply, @!

No, we wish to train using MONAI Label, but we wish to train on a separate machine. That is, the local machine where 3d Slicer is running is where we will perform the annotations, but the monai server will be launched on a separate machine. Is that possible?

**Reply by tangy5** [reply]
Yes. 3D Slicer can link to MONAI Label server on that separate machine.
Can you run both inference and training on same machine (e.g., remote machine)? 

You can set the IP address and port and use 3D Slicer locally for annotation when running MONAI Label server remotely. 

For example: 

```bash
monailabel start_server --app apps/radiology --studies datasets/Task09_Spleen/imagesTr --conf models segmentation_spleen --host <remote machine IP> --port <#port, e.g., 8000>
```


Then when starting 3D Slicer locally:

set IP and port here: e.g., http://{remote IP}:8000

![Screenshot from 2022-12-09 15-40-26](https://user-images.githubusercontent.com/58751975/206812626-2183c528-ed1e-4260-9348-7e4f4e49de5c.png)

**Reply by andreped** [reply]
Just tried your suggestion, @tangy5, and it works great! **Fantastic!**
I had no idea that MONAI Label supported annotating locally and using a separate server for training the models. That's fantastic!

Had some issues with ports on a the server we were testing this on, which we were not able to fix until just now.

However, when testing it now, we observed some strange naming issues after submitting labels twice, where `.nii.gz` where appended twice to the names. But we used an older version of MONAI Label. Will try the new release and report any issues/bugs if there are any.

Great work by @diazandr3s and the team!

The person currently actively testing this from our end is @karenhes, and might be the one asking questions and/or reporting issues, if there are any, instead of me.
